caas/kubernetes/provider/specs/v2.go

```diff
@@ -40,10 +41,16 @@ func fileSetsV2ToFileSets(fs []specs.FileSetV2) (out []specs.FileSet) {
 			Name:      f.Name,
 			MountPath: f.MountPath,
 		}
-		for k, v := range f.Files {
+
+		// We sort the keys of the files here to get a deterministic ordering.
+		// See lp:1895598
+		keys := specs.KeysForFiles(f.Files)
+		sort.Strings(keys)
```

Comment:
If we are using a helper function (perhaps overkill), let's put the sort in there too. And rename to SortedKeysForFiles()